google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -93,12 +97,302 @@ public BucketInfo apply(com.google.api.services.storage.model.Bucket pb) {
+      private final DateTime createdBefore;
+      private final Integer numberOfNewerVersions;
+      private final Boolean isLive;
+      private final List<StorageClass> matchesStorageClass;
+
+      private LifecycleCondition(Builder builder) {
+        this.age = builder.age;
+        this.createdBefore = builder.createdBefore;
+        this.numberOfNewerVersions = builder.numberOfNewerVersions;
+        this.isLive = builder.isLive;
+        this.matchesStorageClass = builder.matchesStorageClass;
+      }
+
+      public Builder toBuilder() {
+        return newBuilder()
+                .setAge(this.age)
+                .setCreatedBefore(this.createdBefore)
+                .setNumberOfNewerVersions(this.numberOfNewerVersions)
+                .setIsLive(this.isLive)
+                .setMatchesStorageClass(this.matchesStorageClass);
+      }
+
+      public static Builder newBuilder() {
+        return new Builder();
+      }
+
+      public Integer getAge() {
+        return age;
+      }
+
+      public DateTime getCreatedBefore() {
+        return createdBefore;
+      }
+
+      public Integer getNumberOfNewerVersions() {
+        return numberOfNewerVersions;
+      }
+
+      public Boolean getIsLive() {
+        return isLive;
+      }
+
+      public List<StorageClass> getMatchesStorageClass() {
+        return matchesStorageClass;
+      }
+
+      /** Builder for {@code LifecycleCondition}. */
+      public static class Builder {
+        private Integer age;
+        private DateTime createdBefore;
+        private Integer numberOfNewerVersions;
+        private Boolean isLive;
+        private List<StorageClass> matchesStorageClass;
+
+        private Builder() {}
+
+        /** Sets the age in days a Blob should be for an Action to be executed **/
+        public Builder setAge(Integer age) {
+          this.age = age;
+          return this;
+        }
+
+        /** Sets the date a Blob should be created before for an Action to be executed. Note that only the date
+         * will be considered, if the time is specified it will be truncated **/
+        public Builder setCreatedBefore(DateTime createdBefore) {
+          this.createdBefore = createdBefore;
+          return this;
+        }
+
+        /** Sets the number of newer versions a Blob should have for an Action to be executed **/
+        public Builder setNumberOfNewerVersions(Integer numberOfNewerVersions) {
+          this.numberOfNewerVersions = numberOfNewerVersions;
+          return this;
+        }
+
+        /** Sets whether a Live Blob meets the Condition for an Action to be executed. If this is set to {@code false},
+         * the condition is met by Archived Blobs.
+         */
+        public Builder setIsLive(Boolean live) {
+          this.isLive = live;
+          return this;
+        }
+
+        /** Sets a list of Storage Classes ({@code StorageClass} objects) for a Blob to match for an Action to be executed **/
```

Comment:
Sets a list of Storage Classes for a objects that satisfy the condition to execute the Action.